contracts/vesting/src/vesting/account/mod.rs

```diff
@@ -124,105 +110,88 @@ impl Account {
-        let balance = Item::new(&key);
-        Ok(balance.save(storage, &amount)?)
+        save_balance(self.storage_key(), amount, storage)
     }
 
     pub fn load_mixnode_pledge(
         &self,
         storage: &dyn Storage,
     ) -> Result<Option<PledgeData>, ContractError> {
-        let key = self.mixnode_pledge_key();
-        let mixnode_pledge = Item::new(&key);
-        Ok(mixnode_pledge.may_load(storage)?)
+        load_bond_pledge(self.storage_key(), storage)
     }
 
     pub fn save_mixnode_pledge(
         &self,
         pledge: PledgeData,
         storage: &mut dyn Storage,
     ) -> Result<(), ContractError> {
-        let key = self.mixnode_pledge_key();
-        let mixnode_pledge = Item::new(&key);
-        Ok(mixnode_pledge.save(storage, &pledge)?)
+        save_bond_pledge(self.storage_key(), &pledge, storage)
     }
 
     pub fn remove_mixnode_pledge(&self, storage: &mut dyn Storage) -> Result<(), ContractError> {
-        let key = self.mixnode_pledge_key();
-        let mixnode_pledge: Item<PledgeData> = Item::new(&key);
-        mixnode_pledge.remove(storage);
-        Ok(())
+        remove_bond_pledge(self.storage_key(), storage)
     }
 
     pub fn load_gateway_pledge(
         &self,
         storage: &dyn Storage,
     ) -> Result<Option<PledgeData>, ContractError> {
-        let key = self.gateway_pledge_key();
-        let gateway_pledge = Item::new(&key);
-        Ok(gateway_pledge.may_load(storage)?)
+        load_gateway_pledge(self.storage_key(), storage)
     }
 
     pub fn save_gateway_pledge(
         &self,
         pledge: PledgeData,
         storage: &mut dyn Storage,
     ) -> Result<(), ContractError> {
-        let key = self.gateway_pledge_key();
-        let gateway_pledge = Item::new(&key);
-        Ok(gateway_pledge.save(storage, &pledge)?)
+        save_gateway_pledge(self.storage_key(), &pledge, storage)
     }
 
     pub fn remove_gateway_pledge(&self, storage: &mut dyn Storage) -> Result<(), ContractError> {
-        let key = self.gateway_pledge_key();
-        let gateway_pledge: Item<PledgeData> = Item::new(&key);
-        gateway_pledge.remove(storage);
-        Ok(())
+        remove_gateway_pledge(self.storage_key(), storage)
     }
 
     // Returns block_time part of the delegation key
-    pub fn delegation_keys_for_mix(&self, mix: &str, storage: &dyn Storage) -> Vec<u64> {
-        let key = self.delegations_key();
-        let delegations: Map<(&[u8], u64), Uint128> = Map::new(&key);
-        delegations
-            .prefix(mix.as_bytes())
-            .keys(storage, None, None, Order::Ascending)
-            // Scan will blow up on first error
-            .scan((), |_, x| x.ok())
-            .collect::<Vec<u64>>()
+    pub fn delegation_block_times_for_mix(
+        &self,
+        mix: &str,
+        storage: &dyn Storage,
+    ) -> Result<Vec<u64>, ContractError> {
+        let delegations = load_delegations_for_mix(self.storage_key(), mix, storage)?;
```

Comment:
Also, how many values do you reckon could be loaded at the most? Because the of usual, it's an unbonded call and all of the associated "nice" stuff could happen...

Reply:
I'm hoping that no one will delegate to a mix a 100 times, I can't imagine what the use case would be, then again, someone might do it just for fun.

Reply:
Does it make a difference that we're not materialising all the values, but iterating and folding?